cores/esp8266/WString.h

```diff
@@ -230,6 +230,33 @@ class String {
         bool operator ==(const __FlashStringHelper *rhs) const {
             return equals(rhs);
         }
+        bool operator ==(std::nullptr_t) const {
+            return length() == 0;
+        }
```

Comment:
re. comment, only this part would make sense for a fix. the rest is even more ambigious issues later down the line